### PR TITLE
Add Renode Sample

### LINKED ISSUE #9: Renode sample and documentation
## Context

Renode is a virtual development tool for multi-node embedded networks (both wired and wireless) and is intended to enable a scalable workflow for creating effective, tested and secure IoT systems.

https://github.com/renode/renode

Renode can take unmodified binaries (that you would run a physical board) and run them in a virtual environment.

On the surface it seems that Renode has a ton of features that would be useful in our embedded development.

## Required steps

* Research what Renode can bring to the table
* Write out Implementation steps

## Definition of Done

Above is done, reviewed and tested.

### PR DESCRIPTION
**Pull request by MarkoSagadin**
## Description

This PR adds a sample that demonstrates Renode's capabilities. Relevant documentation and installation script are also provided.
More is briefly described in the commit message.

Closes #9 #11

## Areas of interest for the reviewer

Read the documentation to see if it is clearly written.

## Checklist

- [x] My code follows the [style guidelines] as defined by IRNAS.
- [x] I have performed a self-review of my code.
- [x] My changes generate no new warnings.
- [x] I added/updated source code documentation for all newly added or changed functions.

## After-review steps

- I will merge PR by myself.

[style guidelines]:
  https://github.com/IRNAS/irnas-guidelines-docs/blob/main/docs/developer_guidelines.md

### CODE REVIEW

**Comment by MarkoSagadin** [comment]
Current dependencies on/for this PR:
* main
  * **PR #12** <a href="https://app.graphite.dev/github/pr/IRNAS/irnas-zephyr-template/12" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/IRNAS/irnas-zephyr-template/12?utm_source=stack-comment).

File: samples/renode_example/README.md [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+the nrf52840dk board.
+
+This sample contains a simple firmware for the nrf52840dk board, which does the
+following:
+
+- On the press of a Button 1 (on the nrf52840DK) it prints
+  `GPIO Interrupt fired`,
+- It has enabled GPIO and Sensor shells and
+- It specifies in its device tree LIS2DW12 accelerometer and STTS751 temperature
+  sensors.
```
Comment (klancarLuka) [inline samples/renode_example/README.md:17]:
```suggestion
- On the press of Button 1 (on the nrf52840DK) it prints
  `GPIO Interrupt fired`.
- It has enabled GPIO and Sensor shells.
- It specifies in its device tree LIS2DW12 accelerometer and STTS751 temperature
  sensors.
```
If the text of your bullet point is a complete sentence (or multiple sentences), use capital letters and punctuation. If your points are not structured as proper sentences, you don’t need to end with punctuation.

**Review by klancarLuka — COMMENTED** [review]
Only one minor correction. :)

File: samples/renode_example/README.md (hunk 1 above)
Comment (MarkoSagadin) [reply samples/renode_example/README.md:17]: Corrected :+1:

File: samples/renode_example/renode/STTS751/README.md [hunk 2]
```diff
@@ -0,0 +1,4 @@
+# STTS751
+
+This document outlines the steps required to implement a custom peripheral for
+Phenode. This chip is supported by Zephyr, but is not implemented in Renode.
```
Comment (NejcKle) [inline samples/renode_example/renode/STTS751/README.md:4]: Add the following:
```
1. Add I2C device to .overlay file - driver must be supported in Zephyr
2. Add sensor to .repl file, using the same address as in .overlay file
3. Add Sensor implementation to STTS751.cs. There are several ways to implement the
sensor. We have followed the example of LIS2DW12.cs. Several methods must be implemented, this
depends on the class the sensor inherits from.
4. Include the STTS751.cs sensor in the .resc file using the `include` syntax
5. Add a simple test to the .robot file

Some useful information regarding the implementation can be found here:
* [Renode - Writing a new peripheral](https://renode.readthedocs.io/en/latest/advanced/writing-peripherals.html#)
* [LIS2DW12 .cs implementation](https://github.com/renode/renode-infrastructure/blob/master/src/Emulator/Peripherals/Peripherals/Sensors/LIS2DW12.cs)
* [Step by Step Implementation guide by betrusted[(https://betrusted.io/xous-book/ch04-04-writing-cs-peripherals.html)
```

File: samples/renode_example/test-nrf52840dk.robot [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+Set Acc Z
+    [Documentation]     Set acceleration on Z axis and read it back
+
+    Start Test
+
+    Execute Command     ${Accelerometer} AccelerationZ 0.1
+    Write Line To Uart  sensor get lis2dw12@19 accel_z
+    Wait For Line On Uart    0.976271    timeout=2
+
+#Set STTS751 Temperature
```
Comment (NejcKle) [inline samples/renode_example/test-nrf52840dk.robot:73]: Why is this commented out? For positive numbers the test works fine :100:

**Review by NejcKle — CHANGES_REQUESTED** [review]

File: samples/renode_example/renode/STTS751/README.md (hunk 2 above)
Comment (MarkoSagadin) [reply samples/renode_example/renode/STTS751/README.md:4]: I moved this text from the top of the STTS751.cs file to this Readme.

File: samples/renode_example/test-nrf52840dk.robot (hunk 3 above)
Comment (MarkoSagadin) [reply samples/renode_example/test-nrf52840dk.robot:73]: Not anymore since the Zephyr driver is wrong.

**Review by NejcKle — APPROVED** [review]

**Pull request merged**